gradle-plugin/src/test-integration/java/io/neow3j/devpack/gradle/Neow3jCompilePluginIntegrationTest.java

```diff
@@ -13,19 +13,76 @@
 import static java.util.Objects.requireNonNull;
 import static org.gradle.testkit.runner.TaskOutcome.FAILED;
 import static org.gradle.testkit.runner.TaskOutcome.SUCCESS;
+import static org.gradle.testkit.runner.TaskOutcome.FROM_CACHE;
+import static org.gradle.testkit.runner.TaskOutcome.UP_TO_DATE;
 import static org.hamcrest.CoreMatchers.not;
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.containsString;
 import static org.hamcrest.Matchers.greaterThan;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
 
 public class Neow3jCompilePluginIntegrationTest {
 
     @TempDir
     public Path projectRootDir;
 
+    @Test
+    public void testTaskCompilationCacheIfFileChanges() throws IOException {
+        String buildFileContent = "" +
+              "sourceCompatibility = JavaVersion.VERSION_1_8" + "\n" +
+              "targetCompatibility = JavaVersion.VERSION_1_8" + "\n" +
+              "\n" +
+              "neow3jCompiler {" + "\n" +
+              "    className=" + "\"io.neow3j.devpack.gradle.ContractTest\"" + "\n" +
+              "}" + "\n";
+
+        GradleProjectTestCase testCase = new GradleProjectTestCase(this.projectRootDir)
+              .withDefaultDependencies()
+              .appendToBuildFile(buildFileContent)
+              .withContractName("ContractTest")
+              .withContractSourceFileName("ContractTest.java")
+              .runBuild();
+
+        BuildResult buildResult = testCase.getGradleBuildResult();
+
+        assertEquals(SUCCESS, buildResult.task(":" + TASK_NAME).getOutcome());
+        testCase = testCase.replaceMethodName("test", "testCached");
+        // run for the second time to assert cache was not used
+        testCase.runBuild();
+        buildResult = testCase.getGradleBuildResult();
+        assertNotEquals(FROM_CACHE, buildResult.task(":" + TASK_NAME).getOutcome());
```

Comment:
Just for clarity, I would add:

```suggestion
        assertNotEquals(FROM_CACHE, buildResult.task(":" + TASK_NAME).getOutcome());
        assertNotEquals(UP_TO_DATE, buildResult.task(":" + TASK_NAME).getOutcome());
```

not strictly required, because the following line has an `assertEquals()`, but just in case, for "maintainability" (if someone reads this test).